packages/atomic/src/components/common/result-list/result-list-common.tsx

```diff
@@ -67,6 +71,29 @@ export class ResultListCommon<SpecificResult extends AnyResult = AnyResult>
     this.props.nextNewResultTarget.focusOnNextTarget();
   }
 
+  public async focusOnFirstResultAfterNextSearch() {
+    await defer();
+    return new Promise<void>((resolve) => {
+      if (this.props.getResultListState().isLoading) {
+        this.firstResultEl = undefined;
+      }
+
+      const unsub = this.props.bindings.engine.subscribe(async () => {
+        await defer();
+        if (!this.props.getResultListState().isLoading && this.firstResultEl) {
+          const elementToFocus =
+            getFirstFocusableDescendant(this.firstResultEl) ??
+            this.firstResultEl;
+          this.props.nextNewResultTarget.setTarget(elementToFocus);
+          this.props.nextNewResultTarget.focus();
+          this.firstResultEl = undefined;
+          unsub();
+          resolve();
+        }
+      });
+    });
+  }
```

Comment:
This is a lot of logic that's separate from the "next new result" logic, I wonder if we could reduce that.

I would especially like to avoid needing to re-subscribe to the engine and `defer` to wait for the state to be updated. I think we should rely on Stencil's re-rendering mechanism instead (just like the focus targets).

Would copy and pasting `focusOnNextNewResult()` like so work?
```ts
public focusOnFirstResultAfterNextSearch() {
  this.indexOfResultToFocus = 0;
  this.props.nextNewResultTarget.focusOnNextTarget();
}
```

Reply:
No, unfortunately this does not work.

This is exactly what I tried first while working on this problem.

All the reference to the various HTML elements changes multiple times as a new query gets executed and the result-list-common does it's thing.

I've realized the `result-list-common` rendering loop is not efficient, and will needlessly re-render multiple times on a new query. 

So the `index=0` thingy does not work since we end trying to focus on an element that will get removed from the DOM.

The reason it works for `focusOnNextNewResult` is because the JSX ID for existing results is stable when we "load more". ie: The rendering loop for existing result when we add more of them is in this case correctly implemented.

This is one thing we need to refactor for `result-list-common`, which is partially why I added that comment in Slack yesterday.